### PR TITLE
 Fix warnings in analyzer packages

### LINKED ISSUE #22090: Fix roslyn warnings in Analyzer packages
There are a few warnings in our analyzer packages that need to be fixed:

```
ComponentFacts.cs(27,24): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(27,74): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(43,54): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(58,74): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
InternalUsageAnalyzer.cs(129,35): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentParameterAnalyzer.cs(30,57): warning RS1026: Enable concurrent execution [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentParameterAnalyzer.cs(30,57): warning RS1025: Configure generated code analysis [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentParameterUsageAnalyzer.cs(26,57): warning RS1026: Enable concurrent execution [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
InternalUsageAnalyzer.cs(158,35): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(87,54): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentInternalUsageDiagnosticAnalzyer.cs(31,57): warning RS1026: Enable concurrent execution [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
```

Currently, these warnings are disabled in https://github.com/dotnet/aspnetcore/blob/b9899d9649d43a3816ecc0f30669551e0492dbcb/Directory.Build.props#L74-L75. We should fix the warnings and stop ignoring these in Directory.Build.props.

FYI @NTaylorMullen to assign since I'm not too sure who would own these.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #22090

### CODE REVIEW

File: src/Mvc/Mvc.Analyzers/src/CodeAnalysisExtensions.cs [hunk 1]
```diff
@@ -97,7 +97,7 @@ public static bool IsAssignableFrom(this ITypeSymbol source, ITypeSymbol target)
             source = source ?? throw new ArgumentNullException(nameof(source));
             target = target ?? throw new ArgumentNullException(nameof(target));
 
-            if (source == target)
+            if (source.Equals(target))
```
Comment (Youssef1313) [inline src/Mvc/Mvc.Analyzers/src/CodeAnalysisExtensions.cs:100]:
```suggestion
            if (SymbolEqualityComparer.Default.Equals(source, target))
```

File: src/Mvc/Mvc.Analyzers/src/CodeAnalysisExtensions.cs [hunk 2]
```diff
@@ -106,7 +106,7 @@ public static bool IsAssignableFrom(this ITypeSymbol source, ITypeSymbol target)
             {
                 foreach (var @interface in target.AllInterfaces)
                 {
-                    if (source == @interface)
+                    if (source.Equals(@interface))
```
Comment (Youssef1313) [inline src/Mvc/Mvc.Analyzers/src/CodeAnalysisExtensions.cs:109]:
```suggestion
                    if (SymbolEqualityComparer.Default.Equals(source, @interface))
```

File: src/Mvc/Mvc.Analyzers/src/CodeAnalysisExtensions.cs [hunk 3]
```diff
@@ -117,7 +117,7 @@ public static bool IsAssignableFrom(this ITypeSymbol source, ITypeSymbol target)
 
             foreach (var type in target.GetTypeHierarchy())
             {
-                if (source == type)
+                if (source.Equals(type))
```
Comment (Youssef1313) [inline src/Mvc/Mvc.Analyzers/src/CodeAnalysisExtensions.cs:120]:
```suggestion
                if (SymbolEqualityComparer.Default.Equals(source, type))
```

File: src/Mvc/Mvc.Analyzers/src/TagHelpersInCodeBlocksAnalyzer.cs [hunk 4]
```diff
@@ -105,7 +106,7 @@ private void InitializeWorker(CompilationStartAnalysisContext context, SymbolCac
 
         private bool IsTagHelperRunnerRunAsync(IMethodSymbol method, SymbolCache symbolCache)
         {
-            if (method != symbolCache.TagHelperRunnerRunAsyncMethodSymbol)
+            if (!method.Equals(symbolCache.TagHelperRunnerRunAsyncMethodSymbol))
```
Comment (Youssef1313) [inline src/Mvc/Mvc.Analyzers/src/TagHelpersInCodeBlocksAnalyzer.cs:109]:
```suggestion
            if (!SymbolEqualityComparer.Default.Equals(method, symbolCache.TagHelperRunnerRunAsyncMethodSymbol))
```

File: src/Mvc/Mvc.Analyzers/src/TopLevelParameterNameAnalyzer.cs [hunk 5]
```diff
@@ -152,7 +152,7 @@ internal static string GetName(in SymbolCache symbolCache, ISymbol symbol)
             foreach (var attribute in symbol.GetAttributes(symbolCache.IModelNameProvider))
             {
                 // BindAttribute uses the Prefix property as an alias for IModelNameProvider.Name
-                var nameProperty = attribute.AttributeClass == symbolCache.BindAttribute ? "Prefix" : "Name";
+                var nameProperty = attribute.AttributeClass.Equals(symbolCache.BindAttribute) ? "Prefix" : "Name";
```
Comment (Youssef1313) [inline src/Mvc/Mvc.Analyzers/src/TopLevelParameterNameAnalyzer.cs:155]:
```suggestion
                var nameProperty = SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, symbolCache.BindAttribute) ? "Prefix" : "Name";
```

File: src/Mvc/Mvc.Analyzers/src/ViewFeaturesAnalyzerContext.cs [hunk 6]
```diff
@@ -32,7 +32,7 @@ public bool IsHtmlHelperExtensionMethod(IMethodSymbol method)
                 return false;
             }
 
-            if (method.ContainingType != HtmlHelperPartialExtensionsType)
+            if (!method.ContainingType.Equals(HtmlHelperPartialExtensionsType))
```
Comment (Youssef1313) [inline src/Mvc/Mvc.Analyzers/src/ViewFeaturesAnalyzerContext.cs:35]:
```suggestion
            if (!SymbolEqualityComparer.Default.Equals(method, HtmlHelperPartialExtensionsType))
```

**Comment by Youssef1313** [comment]
I can't figure out the cause of the current test failure.

**Review by pranavkm — APPROVED** [review]

**Review by captainsafia — APPROVED** [review]
Thanks for taking the time to submit this PR, @Youssef1313! Much appreciated. 🙇

**Pull request merged**